algorithms/CPlusPlus/Arrays/fractional-knapsack.cpp

```diff
@@ -0,0 +1,65 @@
+// Description : Given weights and values of n items, put these items in a knapsack of capacity W to get the maximum total value in the knapsack.
+// We can break items for maximizing the total value of knapsack/
+
+// Algorithm Type: Divide & Conquer
+// Time Complexity: O(n*log(n))
+#include <bits/stdc++.h>
+using namespace std;
+class Item
+{
+public:
+    int value, weight;
+
+    // Constructor
+    Item(int value, int weight)
+    {
+        this->value = value;
+        this->weight = weight;
+    }
+};
+bool comp(Item a, Item b)
+{
+    double r1 = (double)a.value / (double)a.weight;
+    double r2 = (double)b.value / (double)b.weight;
+    return r1 > r2;
+}
+
+//Function to get the maximum total value in the knapsack.
+double fractionalKnapsack(int W, Item arr[], int n)
+{
+
+    sort(arr, arr + n, comp);
+
+    int curWeight = 0;
+    double finalvalue = 0.0;
+
+    for (int i = 0; i < n; i++)
+    {
+
+        if (curWeight + arr[i].weight <= W)

```

Comment:
remove unnecessary spaces

Reply:
Can you point out where exactly have he got additional spacing ? because the code looks fairly readable to me with appropriate indentation

Reply:
Line 30 32 35. Just the unnecessary spaces.

Reply:
Don't get me wrong, but addition of just one line newline character to separate variable initialisation block from rest of the function body to enhance the readability looks totally ok to me.

In fact it is a good code formatting practice as it help visually separate out logical blocks of the code and reduce headaches while debugging the program later down the line. You can try using formatting tools like [clang-format](https://clang.llvm.org/docs/ClangFormat.html) or some online code formatters like [cpp formatter](https://codebeautify.org/cpp-formatter-beautifier) and see that even they approve of this.

Reply:
Okay, no issues! Will keep that in mind!